controller/rest/rest.go

```diff
@@ -2246,3 +2247,33 @@ func getRequestApiVersion(r *http.Request) ApiVersion {
 	}
 	return ApiVersion1
 }
+
+func IsCertNearExpired(certPath string, expireThreshold time.Duration) (bool, error) {
+	certPEM, err := os.ReadFile(certPath)
+	if err != nil {
+		return false, fmt.Errorf("failed to read certificate file %s: %v", certPath, err)
+	}
+
+	// Decode the PEM block
+	block, _ := pem.Decode(certPEM)
+	if block == nil || block.Type != "CERTIFICATE" {
+		return false, errors.New("failed to decode certificate")
+	}
+
+	// Parse the certificate
+	cert, err := x509.ParseCertificate(block.Bytes)
+	if err != nil {
+		return false, fmt.Errorf("failed to parse certificate: %v", err)
+	}
+
+	// Check the expiration date
+	if time.Now().After(cert.NotAfter.Add(-expireThreshold)) {
+		log.WithFields(log.Fields{
+			"expiry":    cert.NotAfter,
+			"threshold": expireThreshold,
+		}).Info("Nearly expired certificate is detected")
```

Comment:
minor: unlike the scenario in cert-upgrader, the function here would be called from time to time.  Maybe consider changing it to Debug level?

Reply:
fixed